frontend/src/metabase/visualizations/visualizations/SmartScalar/compute.js

```diff
@@ -1,206 +1,475 @@
 // eslint-disable-next-line no-restricted-imports -- deprecated usage
-import moment from "moment-timezone";
+import moment from "moment";
 import { t } from "ttag";
 import * as Lib from "metabase-lib";
 import { formatValue } from "metabase/lib/formatting/value";
 import { formatDateTimeRangeWithUnit } from "metabase/lib/formatting/date";
-import { color } from "metabase/lib/colors";
-import { formatChange } from "metabase/visualizations/visualizations/SmartScalar/utils";
+import { color, colors } from "metabase/lib/colors";
+import {
+  COMPARISON_TYPES,
+  formatChange,
+} from "metabase/visualizations/visualizations/SmartScalar/utils";
 import { isEmpty } from "metabase/lib/validate";
 import { isDate } from "metabase-lib/types/utils/isa";
 
-const FALLBACK_DATE_UNIT = "day";
-
 export function computeTrend(series, insights, settings) {
+  // get current metric data
+  const currentMetricData = getCurrentMetricData({
+    series,
+    insights,
+    settings,
+  });
+  if (isEmpty(currentMetricData)) {
+    return null;
+  }
+  const { clicked, date, dateUnitSettings, formatOptions, value } =
+    currentMetricData;
+
+  // get comparison data
+  const comparisonData = computeComparison({
+    currentMetricData,
+    series,
+    settings,
+  });
+  const { comparisonDescStr, comparisonValue } = comparisonData ?? {};
+
+  // find percent change in values
+  const percentChange = !isEmpty(comparisonValue)
+    ? computeChange(comparisonValue, value)
+    : undefined;
+
+  const {
+    changeType,
+    changeArrowIconName,
+    percentChangeStr,
+    comparisonValueStr,
+  } = computeChangeTypeWithOptions({
+    comparisonValue,
+    formatOptions,
+    percentChange,
+  });
+
+  const changeColor = !isEmpty(changeArrowIconName)
+    ? getArrowColor(
+        changeArrowIconName,
+        settings["scalar.switch_positive_negative"],
+      )
+    : undefined;
+
+  const valueStr = formatValue(value, formatOptions);
+  const dateStr = formatDateStr({ date, dateUnitSettings });
+  return {
+    value,
+    clicked,
+    formatOptions,
+    display: {
+      value: valueStr,
+      date: dateStr,
+    },
+    comparison: {
+      changeArrowIconName,
+      changeColor,
+      changeType,
+      comparisonDescStr,
+      comparisonValue,
+      display: {
+        percentChange: percentChangeStr,
+        comparisonValue: comparisonValueStr,
+      },
+      percentChange,
+    },
+  };
+}
+
+function computeComparison({ currentMetricData, series, settings }) {
```

Comment:
@kulyk You'll want to add the new comparison option in this function. The settings will contain the comparison `const { type, label, value } = settings["scalar.comparisons"];`. For the static/custom value comparison, it is mainly checking that value is a number and if so, returning the comparison data as `{ comparisonDescStr: label, comparisonValue: value }`